asset/src/elasticsearch_reader_api/schema.ts

```diff
@@ -1,46 +1,357 @@
+
             if (subsliceByKey) {
-                const configType = apiConfig.type;
                 if (apiVersion <= 5 && (configType == null || !isString(configType) || configType.length === 0)) throw new Error(`For elasticsearch apiVersion ${endpointConfig.apiVersion}, a type must be specified`);
-                if (apiVersion > 5 && (configField == null || !isString(configField) || configField.length === 0)) throw new Error('If subslice_by_key is set to true, the field parameter of the documents must also be set');
+                if (apiVersion > 5 && (id_field_name == null || !isString(id_field_name) || id_field_name.length === 0)) throw new Error('If subslice_by_key is set to true, the id_field_name parameter of the documents must also be set');
+            }
+
+            if (apiConfig.key_range && job.slicers > apiConfig.key_range.length) {
+                throw new Error('The number of slicers specified on the job cannot be more the length of key_range');
+            }
+
+            if (apiConfig.key_type === 'base64url') {
+                if (job.slicers > 64) {
+                    throw new Error('The number of slicers specified on the job cannot be more than 64');
+                }
             }
+
+            if (apiConfig.key_type === 'hexadecimal' || apiConfig.key_type === 'HEXADECIMAL') {
+                if (job.slicers > 16) {
+                    throw new Error('The number of slicers specified on the job cannot be more than 16');
+                }
+            }
+
+            elasticAPI({}, logger).validateGeoParameters(apiConfig);
         });
     }
 
     build(): AnyObject {
-        return clonedSchema;
+        return schema;
+    }
+}
+
+function geoPointValidation(point: string | null):void {
+    if (point) {
+        if (typeof point !== 'string') throw new Error('Invalid geo_point, must be a string IF specified');
+
+        const pieces = point.split(',');
+        if (pieces.length !== 2) throw new Error(`Invalid geo_point, received ${point}`);
+        const latitude = toNumber(pieces[0]);
+        const longitude = toNumber(pieces[1]);
+
+        if (latitude > 90 || latitude < -90) throw new Error(`Invalid latitude parameter, was given ${latitude}, should be >= -90 and <= 90`);
+        if (longitude > 180 || longitude < -180) throw new Error(`Invalid longitude parameter, was given ${longitude}, should be >= -180 and <= 180`);
+    }
+}
+
+function checkUnits(unit: string | null):void {
+    if (unit) {
+        const unitOptions = {
+            mi: true,
+            yd: true,
+            ft: true,
+            km: true,
+            m: true
+        };
+        if (typeof unit !== 'string') throw new Error('Invalid parameter, must be a string IF specified');
+        if (!unitOptions[unit]) throw new Error('Invalid unit type, did not have a proper unit of measurement (ie m, km, yd, ft)');
+    }
+}
+
+function validGeoDistance(distance: string | null):void {
+    if (distance) {
```

Comment:
*SIDE COMMENT* Use return early syntax to clean up nested code blocks, it makes more readable:

```ts
function validGeoDistance(distance: string | null): void {
    if (!distance) return;

    if (typeof distance !== 'string') throw new Error('Invalid geo_distance parameter, must be a string IF specified');
    
    const matches = distance.match(/(\d+)(.*)$/);
    if (!matches) throw new Error('Invalid geo_distance paramter, is formatted incorrectly');

    const number = matches[1];
    if (!number) throw new Error('Invalid geo_distance paramter, it must include a number');

    const unit = matches[2];
    checkUnits(unit);
}
```